src/pybind/rados/rados.pyx

```diff
@@ -1277,6 +1278,41 @@ Rados object in state %s." % self.state)
         if ret < 0:
             raise make_ex(ret, "error blacklisting client '%s'" % client_address)
 
+    def map_object(self, obj_name, pool_id=-1):
+        """
+        Map an object
+
+        :param obj_name: the name of the object to find
+        :type obj_name: str
+        :param pool_id: id of the pool to look up
+        :type pool_id: int
+
+        :raises: :class:`Error`
+        :returns: dict of OSD up/acting set
+        """
+        self.require_state("connected")
+        obj_name =  cstr(obj_name, 'obj_name')
+        cdef:
+            uint32_t _pool_id = pool_id
+            char *_obj_name = obj_name
+            char *up_set = NULL
+            char *acting_set = NULL
+            size_t up_size, acting_size
+
+        try:
+            up_set = <char *>realloc_chk(up_set, 512)
+            acting_set = <char *>realloc_chk(acting_set, 512)
+            with nogil:
+                ret = rados_map_object(self.cluster, _obj_name, up_set, &up_size,
+                                       acting_set, &acting_size, _pool_id)
+
+            if ret < 0:
+                raise make_ex(ret, "error map object '%s'" % obj_name)
+            return ({'up': up_set, 'acting': acting_set})
```

Comment:
seems like it would be easier to consume this api if it returned up_set and acting_set as lists of ints here as well